FWCore/Framework/src/Event.cc

```diff
@@ -114,12 +114,16 @@ namespace edm {
 
   ProcessHistoryID const& Event::processHistoryID() const { return eventPrincipal().processHistoryID(); }
 
-  Provenance Event::getProvenance(BranchID const& bid) const {
-    return provRecorder_.principal().getProvenance(bid, moduleCallingContext_);
+  Provenance Event::getProvenance(BranchID const& bid) const { return provRecorder_.principal().getProvenance(bid); }
+
+  Provenance Event::getProvenance(ProductID const& pid) const { return eventPrincipal().getProvenance(pid); }
```

Comment:
Couldn't these return a const reference as well?
```suggestion
  Provenance const& Event::getProvenance(BranchID const& bid) const { return provRecorder_.principal().getProvenance(bid); }

  Provenance const& Event::getProvenance(ProductID const& pid) const { return eventPrincipal().getProvenance(pid); }
```

Reply:
Returning a copy appears to have been a left over of when the object used to be created by the call. Changed.